user-contrib/Ltac2/g_ltac2.mlg

```diff
@@ -248,6 +249,18 @@ GRAMMAR EXTEND Gram
       | e = ltac1_expr -> { [], e }
     ] ]
   ;
+  tac2expr_in_env :
+    [ [ test_ltac1_env; ids = LIST0 locident; "|-"; e = tac2expr ->
+      { let check { CAst.v = id; CAst.loc = loc } =
+          if Tac2env.is_constructor (Libnames.qualid_of_ident ?loc id) then
+            CErrors.user_err ?loc (Pp.str "Syntax error")
```

Comment:
Can we have a more descriptive message here?

Reply:
Addressed.